### PR TITLE
Add support for pageable operations, ETags

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Updated to the latest TCGC which introduced some codegen changes. Some client names have been fixed, error responses have been removed from the non-error response cases, required constant header values. Also fixed header responses in specific time formats.
Added rush gotest script to run tests for all modules. Fixed error output in scripts when execSync throws an exception. Updated rush to use latest version of pnpm.

### CODE REVIEW

File: packages/typespec-go/test/cadlranch/authentication/apikeygroup/zz_apikey_client.go [hunk 1]
```diff
@@ -32,12 +32,11 @@ func (client *ApiKeyClient) Invalid(ctx context.Context, options *ApiKeyClientIn
 	if err != nil {
 		return ApiKeyClientInvalidResponse{}, err
 	}
-	if !runtime.HasStatusCode(httpResp, http.StatusNoContent, http.StatusForbidden) {
```
Comment (jhendrixMSFT) [inline packages/typespec-go/test/cadlranch/authentication/apikeygroup/zz_apikey_client.go:35]: This is due to the updated TCGC which removed error cases from the non-error responses.

File: packages/typespec-go/test/cadlranch/authentication/apikeygroup/zz_responses.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ package apikeygroup
 
 // ApiKeyClientInvalidResponse contains the response from method ApiKeyClient.Invalid.
 type ApiKeyClientInvalidResponse struct {
-	InvalidAuth
```
Comment (jhendrixMSFT) [inline packages/typespec-go/test/cadlranch/authentication/apikeygroup/zz_responses.go:12]: This is due to the updated TCGC which removed error cases from the non-error responses.

File: packages/typespec-go/test/cadlranch/azure/client-generator-core/coreusagegroup/zz_modelinoperation_client.go [hunk 3]
```diff
@@ -15,9 +15,9 @@ import (
 	"net/http"
 )
 
-// UsageClient contains the methods for the _Specs_.Azure.ClientGenerator.Core.Usage group.
+// ModelInOperationClient contains the methods for the _Specs_.Azure.ClientGenerator.Core.Usage group.
 // Don't use this type directly, use a constructor function instead.
-type UsageClient struct {
+type ModelInOperationClient struct {
```
Comment (jhendrixMSFT) [inline packages/typespec-go/test/cadlranch/azure/client-generator-core/coreusagegroup/zz_modelinoperation_client.go:20]: Updated TCGC fixed the client names.

File: packages/typespec-go/test/cadlranch/encode/bytesgroup/zz_property_client.go [hunk 4]
```diff
@@ -46,6 +46,8 @@ func (client *PropertyClient) base64CreateRequest(ctx context.Context, body Base
 	if err != nil {
 		return nil, err
 	}
+	req.Raw().Header["Content-Type"] = []string{"application/json"}
+	req.Raw().Header["Accept"] = []string{"application/json"}
```
Comment (jhendrixMSFT) [inline packages/typespec-go/test/cadlranch/encode/bytesgroup/zz_property_client.go:50]: TCGC now includes required, constant headers.

File: packages/typespec-go/test/cadlranch/encode/datetimegroup/zz_responseheader_client.go
Comment (jhendrixMSFT) [inline packages/typespec-go/test/cadlranch/encode/datetimegroup/zz_responseheader_client.go:1]: TCGC fixed header time responses.

File: packages/typespec-go/src/tcgcadapter/clients.ts [hunk 5]
```diff
@@ -115,37 +116,41 @@ export class clientAdapter {
...
   private adaptMethodParameter(param: tcgc.SdkBodyParameter | tcgc.SdkHeaderParameter | tcgc.SdkPathParameter | tcgc.SdkQueryParameter, optionalGroup?: go.ParameterGroup): go.Parameter {
+    let location: go.ParameterLocation = 'method';
+    if (param.onClient) {
+      // check if we've already adapted this client parameter
+      // TODO: grouped client params
+      const clientParam = this.clientParams.get(param.nameInClient);
+      if (clientParam) {
+        return clientParam;
+      }
+      location = 'client';
```
Comment (jhendrixMSFT) [inline packages/typespec-go/src/tcgcadapter/clients.ts:151]: Fixed placement of client params.

**Review by chlowell — APPROVED** [review]

**Pull request merged**